.github/actions/a11y-sla-breach-labels/action.yaml

```diff
@@ -0,0 +1,160 @@
+          exit 0
+        fi
+
+        declare -A GROUPED_ISSUES
+
+        # Process issues
+        mapfile -t ISSUES < <(echo "$ALL_ISSUES" | jq -c '.[]')
+
+        for issue in "${ISSUES[@]}"; do
+          CREATED_AT=$(echo "$issue" | jq -r '.created_at')
+          ISSUE_NUMBER=$(echo "$issue" | jq -r '.number')
+
+          # Convert date to readable format
+          CREATED_DATE=$(date -d "$CREATED_AT" +"%Y-%m-%d" 2>/dev/null || date -u -jf "%Y-%m-%dT%H:%M:%SZ" "$CREATED_AT" +"%Y-%m-%d")
+
+          # Calculate issue age in weeks
+          CREATED_TIMESTAMP=$(date -d "$CREATED_AT" +%s 2>/dev/null || date -u -jf "%Y-%m-%dT%H:%M:%SZ" "$CREATED_AT" +%s)
+          DAYS_OLD=$(( (CURRENT_TIMESTAMP - CREATED_TIMESTAMP) / 86400 ))
+          WEEKS_OLD=$(( DAYS_OLD / 7 ))
+
+          # Fetch issue labels
+          LABELS=$(echo "$issue" | jq -r '[.labels[].name] | join(", ")')
+
+          # Determine impact level
+          IMPACT_LEVEL=""
+          for level in "${!LABEL_THRESHOLDS[@]}"; do
+            if [[ "$LABELS" == *"$level"* ]]; then
+              IMPACT_LEVEL="$level"
+              break
+            fi
+          done
+
+          if [[ -z "$IMPACT_LEVEL" ]]; then
+            echo "⚠️ Issue #$ISSUE_NUMBER has no recognized impact level (Blocker, Critical, Serious, Moderate). Skipping."
+            continue
+          fi
+
+          # Define SLA thresholds based on impact
+          IMPACT_SLA=${LABEL_THRESHOLDS[$IMPACT_LEVEL]}
+
+          # Determine the correct SLA label
+          NEW_SLA=""
+          if (( WEEKS_OLD >= IMPACT_SLA )); then
+            NEW_SLA="SLA Breach"
+          elif (( WEEKS_OLD >= IMPACT_SLA - 1 )); then
+            NEW_SLA="SLA P1"
+          elif (( WEEKS_OLD >= IMPACT_SLA - 2 )); then
+            NEW_SLA="SLA P2"
+          elif (( WEEKS_OLD >= IMPACT_SLA - 3 )); then
+            NEW_SLA="SLA P3"
+          fi
+
+          # Check current SLA label
+          CURRENT_SLA=""
+          for sla_label in "${SLA_LABELS[@]}"; do
+            if [[ "$LABELS" == *"$sla_label"* ]]; then
+              CURRENT_SLA="$sla_label"
+              break
+            fi
+          done
+
+          # If SLA P1/P2/P3 exists and it's within range, skip adding SLA Breach
+          if [[ "$NEW_SLA" == "SLA Breach" && "$CURRENT_SLA" != "" && "$CURRENT_SLA" != "SLA Breach" ]]; then
+            echo "✅ Issue #$ISSUE_NUMBER is within SLA ($CURRENT_SLA). No breach applied."
+            continue
+          fi
+
+          # If SLA needs updating, remove old SLA labels and apply the new one
+          if [[ "$NEW_SLA" != "$CURRENT_SLA" ]]; then
+            echo "🔄 Updating SLA label for Issue #$ISSUE_NUMBER: Removing [$CURRENT_SLA] → Adding [$NEW_SLA]"
+
+            # Remove old SLA labels
+            for old_label in "${SLA_LABELS[@]}"; do
+              if [[ "$LABELS" == *"$old_label"* ]]; then
+                echo "🚫 Removing label: $old_label from Issue #$ISSUE_NUMBER"
```

Comment:
This code seems a little redundant. On line 115-121, we loop through to get the `CURRENT_SLA`, but then we loop again here. It seems like we should only need to do one or the other loops--not both